pkg/kubectl/cmd/apply_test.go

```diff
@@ -550,6 +551,56 @@ func TestApplyRetry(t *testing.T) {
 	}
 }
 
+func TestApplyNoChange(t *testing.T) {
+	initTestErrorHandler(t)
+	nameRC, currentRC := readAndAnnotateReplicationController(t, filenameRC)
+	pathRC := "/namespaces/test/replicationcontrollers/" + nameRC
+
+	patchCount := 0
+	f, tf, _, _ := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.UnstructuredClient = &fake.RESTClient{
+		APIRegistry:          api.Registry,
+		NegotiatedSerializer: unstructuredSerializer,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathRC && m == "GET":
+				file, err := os.Open(filenameRCAnnotationJSON)
+				if err != nil {
+					t.Fatal(err)
+				}
+				body := ioutil.NopCloser(file)
+				return &http.Response{StatusCode: 200, Header: defaultHeader(), Body: body}, nil
+			case p == pathRC && m == "PATCH":
+				patchCount++
+				validatePatchApplication(t, req)
+				bodyRC := ioutil.NopCloser(bytes.NewReader(currentRC))
+				return &http.Response{StatusCode: 200, Header: defaultHeader(), Body: bodyRC}, nil
+			default:
+				t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
+				return nil, nil
+			}
+		}),
+	}
+	tf.Namespace = "test"
```

Comment:
This setup looks like copy-pasted code, please extract to a helper function.

Reply:
A little different.

Reply:
ok